app/src/main/java/io/gnosis/safe/ErrorHandler.kt

```diff
@@ -134,14 +147,19 @@ fun Throwable.toError(): Error =
         // Network-related errors
         this is HttpException -> {
             this.let {
+                val errorBodyString = it.response()?.errorBody()?.string()
                 when {
-                    this.code() == HttpCodes.BAD_REQUEST -> Error.Error400
+                    this.code() == HttpCodes.BAD_REQUEST -> return if (errorBodyString == "[\"Cloud messaging token is linked to another device\"]") {
```

Comment:
Is it correct that this state is only possible if we don't save deviceUuid after registration? With assumption that correct deviceUuid is always saved, we might not have to deal with this error. Checking body string doesn't look very reliable to me.

Reply:
We will have to deal with this, because there are app instances out there that might have this problem. When users upgrade, it should be fixed for good. 

I am happy to check anything here, when we have a specification from the backend. I asked Uxio, and he said there is only the source code.